### PR TITLE
QSP-13 Only Allow Snappy P2P Encoding

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Currently, we allow for both snappy and non-snappy network encodings, which can lead to network partitions if someone has wrong configurations of their node or accidentally messes with the p2p encoding flags.

**Which issues(s) does this PR fix?**

Part of #6327

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Note: This is being discussed in the specs repo as well, but looks like this is the right direction.

https://github.com/ethereum/eth2.0-specs/issues/1931

**Comment by nisdas** [comment]
I think we should hold off on this until is final in the spec, it is still under discussion and hasnt been removed yet

**Comment by rauljordan** [comment]
@nisdas in the eth2 call there was a discussion about this and teams agreed it is fine to remove. It is already removed from gossip in the spec and Danny has mentioned there is no other reason to keep it. The eth2 call was the chance for someone to speak up about it, but no one brought it up

**Comment by nisdas** [comment]
@rauljordan sounds good then, please fix conflicts here

File: beacon-chain/p2p/service.go [hunk 1]
```diff
@@ -261,15 +261,7 @@ func (s *Service) Started() bool {
...
-	encoding := s.cfg.Encoding
-	switch encoding {
-	case encoder.SSZ:
-		return &encoder.SszNetworkEncoder{}
-	case encoder.SSZSnappy:
-		return &encoder.SszNetworkEncoder{UseSnappyCompression: true}
-	default:
-		panic("Invalid Network Encoding Flag Provided")
-	}
+	return &encoder.SszNetworkEncoder{}
```
Comment (0xKiwi) [inline beacon-chain/p2p/service.go:264]: You don't need to set `UseSnappyCompression` true here?

**Pull request merged**